datatypes/src/main/java/org/hyperledger/besu/datatypes/Transaction.java

```diff
@@ -183,4 +187,126 @@ default Optional<? extends Quantity> getMaxFeePerBlobGas() {
    * @return the type of the transaction
    */
   TransactionType getType();
+
+  /**
+   * Boolean which indicates the transaction has associated cost data, whether gas price or 1559 fee
+   * market parameters.
+   *
+   * @return whether cost params are present
+   */
+  boolean hasCostParams();
+
+  /**
+   * Returns the number of blobs this transaction has, or 0 if not a blob transaction type
+   *
+   * @return return the count
+   */
+  int getBlobCount();
+
+  /**
+   * Returns the signature used to sign the transaction.
+   *
+   * @return the signature used to sign the transaction
+   */
+  SECPSignature getSignature();
+
+  /**
+   * Returns the public key extracted from the signature.
+   *
+   * @return the public key
+   */
+  Optional<String> getPublicKey();
+
+  /**
+   * Returns whether the transaction is a contract creation
+   *
+   * @return {@code true} if this is a contract-creation transaction; otherwise {@code false}
+   */
+  boolean isContractCreation();
+
+  /**
+   * Return the maximum fee per gas the sender is willing to pay for this transaction.
+   *
+   * @return max fee per gas in wei
+   */
+  Wei getMaxGasPrice();
+
+  /**
+   * Return the effective priority fee per gas for this transaction.
+   *
+   * @param maybeBaseFee base fee in case of EIP-1559 transaction
+   * @return priority fee per gas in wei
+   */
+  Wei getEffectivePriorityFeePerGas(final Optional<Wei> maybeBaseFee);
+
+  /**
+   * Return the versioned hashes for this transaction.
+   *
+   * @return optional list of versioned hashes
+   */
+  Optional<List<VersionedHash>> getVersionedHashes();
+
+  /**
+   * Return the blobs with commitments for this transaction.
+   *
+   * @return optional blobs with commitments
+   */
+  Optional<BlobsWithCommitments> getBlobsWithCommitments();
+
+  /**
+   * Return the address of the contract, if the transaction creates one
+   *
+   * @return address of new contract or empty otherwise
+   */
+  Optional<Address> contractAddress();
+
+  /**
+   * Return the access list in case of EIP-2930 transaction
+   *
+   * @return optional access list
+   */
+  Optional<List<AccessListEntry>> getAccessList();
+
+  /**
+   * Writes the transaction to RLP
+   *
+   * @param out the output to write the transaction to
+   */
+  void writeTo(final RLPOutput out);
```

Comment:
One of my goals is to remove the RLP dependency from the datatypes module.

Reply:
I've replaced it with a new generic method `Bytes encoded();`: [see here](https://github.com/hyperledger/besu/blob/12b88e1550988a9eb934460830fc469f0c87adb8/datatypes/src/main/java/org/hyperledger/besu/datatypes/Transaction.java#L221)

Reply:
After diving into what it would take to remove it I'm thinking it should come back.

Because the CREATE operation depends on the RLP definitions to get the contract addresses the EVM module will still require the RLP, it's not just coming into EVM as a tag-along of datatypes.  So removing RLP from datatypes is not really needed, as some of the RLP dependencies once it is determined they should stay are best handled where they are at design-wise.

For performance we will want to use a passed in RLPOutput object as well.